File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.c
double findMaxAverage(int* nums, int numsSize, int k) {
    int sum = 0;
    
    for (int i = 0; i < k; i++) {
        sum += nums[i];
    }
    
    int max_sum = sum;
    
    for (int i = k; i < numsSize; i++) {
        sum += nums[i] - nums[i - k]; // Add incoming element, remove outgoing element
        if (sum > max_sum) {
            max_sum = sum;
        }
    }
    
    return (double)max_sum / k;
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        n = len(nums)
        j, s = 0, 0
        for i in range(k):
            s += nums[i]

        max_avg = float(s)/k 
        for i in range(k, n):
            s = s - nums[j] + nums[i]
            j += 1
            if max_avg < float(s)/k:
                max_avg = float(s)/k
        return max_avg
            